spras/analysis/ml.py

```diff
@@ -54,9 +54,10 @@ def summarize_networks(file_paths: Iterable[Union[str, PathLike]]) -> pd.DataFra
                     elif direction == "D":
                         # node order does matter for directed edges
                         edges.append(DIR_CONST.join([node1, node2]))
-                    else:
-                        ValueError(f"direction is {direction}, rather than U or D")
-
+                    elif direction != 'Direction' and direction != 'U' and direction != 'D':
+                        raise ValueError(f"direction is {direction}, rather than U or D")
+                elif len(parts) > 4 or (len(parts) < 4 and len(parts) > 0): # empty line in file is okay
```

Comment:
Does this logic simplify to `len(parts) != 0`? The length has to be non-negative, and if we are at this line we know it is not 4.

Reply:
We talked about having an initial step to process the header row. Then we continue and interate, processing the rest of the edges and requiring there are always 4 parts.